JADE.io (Australia).js

```diff
@@ -0,0 +1,819 @@
+		}
+	}
+	if (citationsHere && citationsHere.length) {
+		applyParallel(pickBestLawReport(citationsHere), citationsHere);
+	}
+	else {
+		const summaryURL = findSummaryURL(doc, url);
+		if (summaryURL) {
+			ZU.processDocuments(summaryURL, function (sd) {
+				// Capture MNC if still missing
+				if (!parsed) {
+					const nc2 = findNeutralCitation(sd);
+					if (nc2) {
+						item.court = item.court || nc2.courtAcronym;
+						item.docketNumber = item.docketNumber || nc2.number;
+						if (!ncBareStr) ncBareStr = `[${nc2.year}] ${nc2.courtAcronym} ${nc2.number}`;
+						if (!item.dateDecided) item.dateDecided = nc2.year;
+					}
+				}
+				// Update case name from summary if missing/generic
+				const cn2 = extractCaseName(sd);
+				if (cn2 && (!caseNameStr || isGenericCaseTitle(caseNameStr) || !/(?:\bv\.?\b|\bversus\b)/i.test(caseNameStr))) {
+					item.caseName = cn2;
+					caseNameStr = cn2;
+				}
+				const list = findLawReportCitations(sd);
+				if (list && list.length) applyParallel(pickBestLawReport(list), list);
+				finalize();
+			});
+			return;
+		}
+	}
+
+	// Try to improve decision date from page metadata/content if we only have a year
+	(function () {
+		if (item.dateDecided && /^(19|20)\d{2}$/.test(item.dateDecided) === false) return;
+		let d = null;
+		// Prefer authoritative decision date hints; avoid generic publish/update timestamps
+		d = d || getMeta(doc, "citation_date");
+		d = d || getMeta(doc, "DC.date");
+		d = d || getMeta(doc, "dcterms.date");
+		// time elements sometimes contain the decision date near the heading
+		d = d || ZU.xpathText(doc, "//h1//time/@datetime")
+			|| ZU.xpathText(doc, "//h1//time/text()")
+			|| ZU.xpathText(doc, "//time/@datetime")
+			|| ZU.xpathText(doc, "//time/text()");
+
+		// Scan typical header/metadata containers for an AU-style date
+		if (!d) {
+			const candidates = doc.querySelectorAll("header, h1, h2, .metadata, .meta, .case-metadata, .article-info, .title, .heading, dl, .dl-horizontal");
+			const re = /(\d{1,2})\s+([A-Za-z]+)\s+([12]\d{3})/;
+			for (const el of candidates) {
+				const m = (el.textContent || "").match(re);
+				if (m) {
+					d = m[0];
+					break;
+				}
+			}
+		}
+
+		if (d) {
+			const nd = normaliseDate(d);
+			// If we have an MNC year, require the same year to avoid site publish dates
+			const yFromMNC = (parsed && parsed.year) ? String(parsed.year) : null;
+			const yFromDate = (nd && nd.match(/^(\d{4})/)) ? RegExp.$1 : null;
+			if (!yFromMNC || (yFromDate && yFromDate === yFromMNC)) {
+				item.dateDecided = nd;
+			}
+		}
+	})();
+
+	function finalize() {
+		// Build Extra using the collected case name, MNC, and parallels
+		if (ncBareStr) {
+			const cn = (caseNameStr && !isGenericCaseTitle(caseNameStr)) ? caseNameStr : (item.caseName || '');
+			const segs = parallelSegments.length ? `; ${parallelSegments.join('; ')}` : '';
+			item.extra = `neutralCitation: ${ncBareStr}\nfullNeutralCitation: ${cn ? `${cn} ${ncBareStr}` : ncBareStr}${segs}`;
```

Comment:
Values in extra should be spaced and title-cased (`Neutral Citation`). But actually, these should probably be child notes (`item.notes.push({ note: 'whatever' })`).